### PR TITLE
make @BigQueryType.toTable usable for input #24

### PR DESCRIPTION
**Pull request by nevillelyh**

### CODE REVIEW

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/SchemaUtil.scala [hunk 1]
```diff
@@ -23,15 +23,18 @@ import com.google.api.services.bigquery.model.{TableFieldSchema, TableSchema}
...
+  /** Convert schema to case class definitions. */
   def toPrettyString(schema: TableSchema, name: String, indent: Int): String = {
+    "@BigQueryType.toTable\n" +
     getCaseClass(schema.getFields, name, indent)
   }
 
   private def getRawType(tfs: TableFieldSchema, indent: Int): (String, Seq[String]) = {
     val name = tfs.getType match {
-      case "INTEGER" => "Int"
+      case "INTEGER" => "Long"
```
Comment (ravwojdyla) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/SchemaUtil.scala]: Why is this needed, I probably missed something.

Comment (nevillelyh) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/SchemaUtil.scala]: It's a bug in the original code. INTEGER in BQ actually means Long, and their FLOAT really is Double.

File: scio-core/src/main/scala/com/spotify/scio/experimental/package.scala [hunk 2]
```diff
@@ -77,7 +77,12 @@ package object experimental {
         val query = if (newSource != null) newSource else bqt.query.get
         self.bigQuerySelect(query).map(bqt.fromTableRow)
       } else {
-        throw new IllegalArgumentException(s"Missing table or query field in companion")
+        if (newSource != null) {
+          self.bigQuerySelect(newSource).map(bqt.fromTableRow)
```
Comment (ravwojdyla) [inline scio-core/src/main/scala/com/spotify/scio/experimental/package.scala]: `bqt` would not be derived from `table` nor `query` if it was manually created case class, and that is the use case this is solving - correct? In that case maybe a comment inside the code?

Should the doc above be fixed from:
```
sc.typedBigQuery[Row]("myproject:samples.gsod")
```
to
```
sc.typedBigQuery[Row]("select field from myproject:samples.gsod limit 1000")
```?

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/SchemaUtil.scala (hunk 1 above)
Comment (ravwojdyla) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/SchemaUtil.scala]: Cool 👍

File: scio-core/src/main/scala/com/spotify/scio/experimental/package.scala [hunk 3]
```diff
@@ -176,7 +181,12 @@ package object experimental {
         val query = if (newSource != null) newSource else bqt.query.get
         self.getQueryRows(query).map(bqt.fromTableRow)
       } else {
-        throw new IllegalArgumentException(s"Missing table or query field in companion")
+        if (newSource != null) {
```
Comment (ravwojdyla) [inline scio-core/src/main/scala/com/spotify/scio/experimental/package.scala]: Same as above doc might need to be updated to include query instead of table.

File: scio-core/src/main/scala/com/spotify/scio/experimental/package.scala (hunk 2 above)
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/experimental/package.scala]: Good point. I updated the docs and behavior so `newSource` can be either table or query if `query` or `table` is missing form companion object.

File: scio-core/src/main/scala/com/spotify/scio/experimental/package.scala (hunk 3 above)
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/experimental/package.scala]: Updated doc and behavior.

File: scio-core/src/main/scala/com/spotify/scio/experimental/package.scala [hunk 4]
```diff
@@ -164,20 +176,33 @@ package object experimental {
      *
      * // Read from [myproject:samples.gsod] instead.
      * bq.getTypedRows[Row]("myproject:samples.gsod")
+     *
+     * // Read from a query instead.
+     * sc.getTypedRows[Row]("SELECT * FROM [publicdata:samples.gsod] LIMIT 1000")
```
Comment (ravwojdyla) [inline scio-core/src/main/scala/com/spotify/scio/experimental/package.scala]: Everything looks 👍 the only thing is that the example should really be a table if it does `SELECT * ...`, I know it's tiny, but even BQ doc says in most cases you should not do `SELECT * ...`

**Comment by ravwojdyla** [comment]
👍
